tests/unit/test_navigation_table.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Unit tests for navigation table module."""
+
+# Need access to protected functions for testing
+# pylint: disable=protected-access
+
+import pytest
+
+from src import navigation_table
+from src.exceptions import NavigationTableParseError
+
+
+@pytest.mark.parametrize(
+    "line, expected_result",
+    [
+        pytest.param("", True, id="empty"),
+        pytest.param("unmatched line", True, id="does not match any line regex"),
+        pytest.param("||||", True, id="line with nothing"),
+        pytest.param("|level|path|navlink|", True, id="matches the header lower case"),
+        pytest.param("|LEVEL|PATH|NAVLINK|", True, id="matches the header upper case"),
+        pytest.param("|Level|Path|Navlink|", True, id="matches the header mixed case"),
+        pytest.param(" |level|path|navlink|", True, id="matches the header single leading space"),
+        pytest.param(
+            "  |level|path|navlink|", True, id="matches the header multiple leading space"
+        ),
+        pytest.param("\t|level|path|navlink|", True, id="matches the header leading tab"),
+        pytest.param("| level|path|navlink|", True, id="matches the header space before level"),
+        pytest.param("|level |path|navlink|", True, id="matches the header space after level"),
+        pytest.param("|level| path|navlink|", True, id="matches the header space before path"),
+        pytest.param("|level|path |navlink|", True, id="matches the header space after path"),
+        pytest.param("|level|path| navlink|", True, id="matches the header space before navlink"),
+        pytest.param("|level|path|navlink |", True, id="matches the header space after navlink"),
+        pytest.param("|level|path|navlink| ", True, id="matches the header trailing space"),
```

Comment:
The reason for that is because these are white box tests and all the whitespace regular expressions use the same variable `_WHITESPACE = r"\s*"`. This means that the code just needs one test case to check that regular expression correctly captures multiple whitespaces, then all that is needed is a check that whitespace checks are where they are expected. Otherwise this list of test would get very long indeed